test/00_transform/MatMul.cu

```diff
@@ -561,3 +561,107 @@ TYPED_TEST(MatMulTestFloatNonHalfTypes,  MatMulOp)
 
   MATX_EXIT_HANDLER();
 }
+
+TYPED_TEST(MatMulTestFloatTypes, MediumMatVec)
+{
+  MATX_ENTER_HANDLER();
+  constexpr index_t m = 128;
+  constexpr index_t k = 256;
+  constexpr index_t n = 1;
+
+  tensor_t<TypeParam, 2> a{{m, k}};
+  tensor_t<TypeParam, 2> b{{k, n}};
+  tensor_t<TypeParam, 2> c{{m, n}};
+  this->pb->template InitAndRunTVGenerator<TypeParam>(
+      "00_transforms", "matmul_operators", "run", {m, k, n});
+
+  this->pb->NumpyToTensorView(a, "a");
+  this->pb->NumpyToTensorView(b, "b");
+
+  auto cs = c.template Slice<1>({0,0}, {matxEnd, matxDropDim});
```

Comment:
you can use `slice` here instead since it's the new preferred method and gets rid of the need for the template keyword

Reply:
@cliffburdick Using `auto cs = slice({0,0}, {matxEnd, matxDropDim});`? Will that result in a rank 2 tensor `cs` with `Size(1) == 0`? In this case, the subsequent `matvec` requires that `b` and `c` both have rank one smaller than `a`. Or perhaps we can generalize `slice` to rank-reduce when using `matxDropDim`.

Reply:
the first parameter to `slice` is the input operator. other than that, the arguments are the same. it should result in exactly the same thing you had, but the syntax is nicer and we're trying to move away from the member functions

Reply:
Oh, sorry, I meant `auto cs = slice(c, {0,0}, {matxEnd, matxDropDim});`. But I think that will give use the following:

Reply:
I believe you'll need to use the [matx::slice with Rank as a template param](https://github.com/NVIDIA/MatX/blob/522c7419648f4a4ca339a6234ba5adda52393882/include/matx/operators/slice.h#L247), as the slice with no templates params will always return with the same rank:

```
    auto c = matx::make_tensor<float,2>({16,16});
    auto cs = matx::slice<1>(c, {0,0}, {matx::matxEnd, matx::matxDropDim});
    std::cout << "Rank of CS: " << cs.Rank() << std::endl;  
```
 output:
 
```
 docker@e5c559bbda66:/home/tallen/scratch/playground/matxExample/build$ ./matxExample 
Rank of CS: 1
```

Reply:
@tylera-nvidia Thanks, I missed that `slice` still had a rank template parameter. I have updated the tests to use `slice` instead of `Slice`.